### ISSUE TITLE
Full calendar overlap background color issue

### ISSUE THREAD

**Issue by maithilir** [fullcalendar/fullcalendar#4271]
Hi,

I am using FullCalendar paid version into my project.
There is a requirement where we need to overlap the background color with the other color.
These colors are generated dynamically and and are based on conditions.

Issue :
Actual issue is these colors are not overlaping. I can see both the colors in for same date and time.
i have tried using overlap for the event but looks like something is missing.

Code snippet:
Below code snippet is from fiddle where i have added a chunk at the end of the code where i have tried to add 2 colors to same date with overlap true and false.
Here i have tried to add the black color and "#ff9f89" color to specified date. and based on overlap feature want to check which color is overlaping.

URL for fiddle : https://codepen.io/pen/?editors=0010

$(function() {

$('#calendar').fullCalendar({
header: {
left: 'prev,next today',
center: 'title',
right: 'month,agendaWeek'
},
defaultDate: '2018-07-12',
events: [
{
start: '2018-07-11T10:00:00',
end: '2018-07-11T16:00:00',
rendering: 'background'
},
{
start: '2018-07-13T10:00:00',
end: '2018-07-13T16:00:00',
rendering: 'background'
},
{
start: '2018-07-24',
end: '2018-07-28',
overlap: false,
rendering: 'background',
color: '#ff9f89'
},
{
start: '2018-07-06',
end: '2018-07-08',
overlap:true,
rendering: 'background',
color: '#ff9f89'
},{
start: '2018-08-04T01:00:00',
end: '2018-08-04T23:00:00',
overlap:true,
rendering: 'background',
color: '#ff9f89'
},{
start: '2018-08-04T07:00:00',
end: '2018-08-04T18:00:00',
overlap: false,
rendering: 'background',
color: 'black'
}
]
});

});

**Comment by maithilir** [comment]
Hi,

I am using FullCalendar paid version into my project.
There is a requirement where we need to overlap the background color with the other color.
These colors are generated dynamically and and are on based on conditions. 

Issue : 
Actual issue is these colors are not overlaping. I can see both the colors in for same date and time. 
i have tried using overlap for the event but looks like something is missing.

Code snippet:
Below code snippet is from fiddle where i have added a chunk at the end of the code where i have tried to add 2 colors to same date with overlap true and false.
URL for fiddle : https://codepen.io/pen/?editors=0010
$(function() {

  $('#calendar').fullCalendar({
    header: {
      left: 'prev,next today',
      center: 'title',
      right: 'month,agendaWeek'
    },
    defaultDate: '2018-07-12',
    events: [
      {
        start: '2018-07-11T10:00:00',
        end: '2018-07-11T16:00:00',
        rendering: 'background'
      },
      {
        start: '2018-07-13T10:00:00',
        end: '2018-07-13T16:00:00',
        rendering: 'background'
      },
      {
        start: '2018-07-24',
        end: '2018-07-28',
        overlap: false,
        rendering: 'background',
        color: '#ff9f89'
      },
      {
        start: '2018-07-06',
        end: '2018-07-08',
        overlap:true,
        rendering: 'background',
        color: '#ff9f89'
      },{
        start: '2018-08-04T01:00:00',
        end: '2018-08-04T23:00:00',
        overlap:true,
        rendering: 'background',
        color: '#ff9f89'
      },{
        start: '2018-08-04T07:00:00',
        end: '2018-08-04T18:00:00',
        overlap: false,
        rendering: 'background',
        color: 'black'
      }
    ]
  });

});

**Comment by snehagxp** [comment]
The color are opaque right now, that causes the background to be shown.

**Comment by acerix** [comment]
The [eventOverlap](https://fullcalendar.io/docs/eventOverlap) option determines if events being dragged and resized are allowed to overlap each other which doesn't sound like what you are looking for.

Looking at August 4, the background events colors overlap as I expect.  How are you expecting it to look?

**Comment by snehagokarn** [comment]
With our functionality, we were planning to show that dates on the calendar are blocked by changing the background colour. There are 2 ways that they get blocked and sometimes might overlap each other due to conditions, as the colours are opaque, they cause them to come on top of each other and form a different colour altogether, we were using overlap false to avoid this.
Do you have any suggestions?

**Comment by acerix** [comment]
The background events are not opaque by default, you could make them opaque with CSS:

https://codepen.io/acerix/pen/wxQeoP?editors=0110

**Comment by acerix** [comment]
You could try eventRender to control which event is rendered when there is an overlap.

You could also request a new feature that works as you describe.  I think it would make more sense as a global option (instead of setting on each event), then maybe something like "z-index" on the events to control which one is on top when there is an overlap.

**Comment by acerix** [comment]
Closing since there is a new issue for the feature request.
